spec/ddtrace/contrib/concurrent_ruby/integration_test_spec.rb

```diff
@@ -27,6 +27,11 @@
     outer_span.finish
   end
 
+  after(:all) do
+    Concurrent.global_io_executor.shutdown
+    Concurrent.global_io_executor.wait_for_termination(5)
+  end
```

Comment:
I'm slightly concerned about this one since it "poisons the well" for any other tests we make that may indirectly use the `global_io_executor`, e.g.:

```ruby
require 'concurrent'

Concurrent.global_io_executor.shutdown
Concurrent.global_io_executor.wait_for_termination
Concurrent.global_io_executor.post { puts 'hello' }
```

```
Traceback (most recent call last):
	6: from concurrent.rb:5:in `<main>'
	5: from /Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/executor/ruby_executor_service.rb:19:in `post'
	4: from /Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:41:in `synchronize'
	3: from /Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:41:in `synchronize'
	2: from /Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:41:in `block in synchronize'
	1: from /Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/executor/ruby_executor_service.rb:21:in `block in post'
/Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby/concurrent/executor/abstract_executor_service.rb:87:in `handle_fallback': Concurrent::RejectedExecutionError (Concurrent::RejectedExecutionError)
```

I don't think concurrent-ruby expects this executor to ever go the way of the dodo.

Reply:
Makes sense. It seems like once stopped it can't be restarted. I'm grouping these threads now, instead of shutting them down.